workers/unity/Packages/com.improbable.gdk.core/Systems/CleanReactiveComponentsSystem.cs

```diff
@@ -54,32 +53,40 @@ private void GenerateComponentGroups()
                         continue;
                     }
 
-                    if (typesToRemove.Contains(type))
+                    if (typesToRemove.Add(type))
```

Comment:
Minor point - while correct this is less clear than what it replaces.

Reply:
It doesn't replace, this returns false if the set already contains the type